os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -221,11 +221,24 @@ void binary_manager_add_binlist(FAR struct tcb_s *tcb)
  ****************************************************************************/
 void binary_manager_remove_binlist(FAR struct tcb_s *tcb)
 {
+	int bin_idx;
 	struct tcb_s *prev;
 	struct tcb_s *next;
 
-	prev = tcb->bin_blink;
-	next = tcb->bin_flink;
-	if (prev) prev->bin_flink = next;
-	if (next) next->bin_blink = prev;
+	bin_idx = tcb->group->tg_binidx;
```

Comment:
Isn't it necessary to check the null pointer of tcb and tcb->group?

Reply:
Yes. it is called after verifying group creation.